### ISSUE TITLE
`filename-case`: `camelCase` should (optionally?) allow uppercase abbreviations like DOM

### ISSUE THREAD

**Issue by thorn0** [sindresorhus/eslint-plugin-unicorn#2141]
There are two styles of camel case used in the wild:

1. Uppercase letters are used only to start words. `getHtml`, `convertFromDom`
2. Abbreviations like DOM or HTML are written all uppercase unless they're the first word: `innerHTML`, `getDOMRangeRect`, `domSelection`

Currently the `filename-case` rule doesn't support the style 2 even though this style is used in the DOM APIs and is overall quite popular.

Example: https://github.com/facebook/lexical/blob/main/packages/lexical-playground/src/utils/getDOMRangeRect.ts

**Comment by dvislov** [comment]
Yes, I also have in the project files with abbreviations inside like `API`, `KPI`, `MUI` (material-ui) etc.

**Comment by fregante** [comment]
I think this is generally called `camelCase` vs `strictCamelCase` (which is what unicorn uses)

The problem with the non-strict version is that it allows eyesores like `XMLHttpRequest`
